### PR TITLE
tablecodec: Add EncodeRow and DecodeRow

### PR DESCRIPTION
**Pull request by shenli**
@coocood

### CODE REVIEW

File: tablecodec/tablecodec.go [hunk 1]
```diff
@@ -193,6 +209,40 @@ func DecodeColumnValue(data []byte, ft *types.FieldType) (types.Datum, error) {
...
+// Row layout: colID1, value1, colID2, value2, .....
+func DecodeRow(data []byte, cols map[int64]*types.FieldType) (map[int64]types.Datum, error) {
+	if data == nil {
+		return nil, nil
+	}
+	values, err := codec.Decode(data)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if len(values) != 2*len(cols) {
```
Comment (coocood) [inline tablecodec/tablecodec.go]: Values may contain unknown column, which has been dropped in a previous DDL.

File: tablecodec/tablecodec_test.go [hunk 2]
```diff
@@ -42,3 +44,88 @@ func (s *testTableCodecSuite) TestTableCodec(c *C) {
...
+	// Compare decoded row and original row
+	for i, col := range cols {
+		v, ok := r[col.id]
+		c.Assert(ok, IsTrue)
+		equal, err1 := v.CompareDatum(row[i])
+		c.Assert(err1, IsNil)
+		c.Assert(equal, Equals, 0)
+	}
+
+	// colMap may contains more columns that encoded row.
```
Comment (disksing) [inline tablecodec/tablecodec_test.go]: ... more columns **than** encoded row.

**Comment by coocood** [comment]
LGTM

**Comment by disksing** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**